nntrainer/tensor/manager.h

```diff
@@ -191,14 +193,16 @@ class Manager {
    * @param weights_spec Specification for the weights
    * @param trainable make the weight trainable if true
    * @param shared_names name to refer to when the weights are borrowed from the
-   * original source. if not shared pass empty vector
+   * @param act_type activation data type if data type is different from weight,
+   * otherwise NONE. original source. if not shared pass empty vector
    *
    * @return created weights list
    */
   std::vector<Weight *>
   requestWeights(const GraphNode &node,
                  const std::vector<Weight::Spec> &weights_spec, bool trainable,
-                 const std::vector<std::string> &shared_names);
+                 const std::vector<std::string> &shared_names,
+                 TensorDim::DataType act_type = TensorDim::DataType::NONE);
```

Comment:
I don't quite understand why a `NONE` data type is added to check that the activation data type is the same as the weight.
The normal activation data type can be passed in while it is the same as weight.
Are there other use cases of `DataTye::NONE` besides this function or have a critical reason to add the NONE data type?

Reply:
It is used only when the master weight is necessary. It's used  to checks activation type is different, and generates master weight.